### DISCUSSION TITLE
RPC Rate Limiting

### DISCUSSION THREAD

**Discussion by agolajko** [zkSync-Community-Hub/zksync-developers#614] [Announcements]
### TL;DR
![image (17)](https://github.com/zkSync-Community-Hub/zksync-developers/assets/22985657/28a051a2-97e4-406b-8c65-b1c744cffe71)

# RPC Rate Limiting

Due to significant community adoption over the past months, we've seen increased requests to our Main Node. To manage this, we are implementing measures to limit excessive API usage, including rate limiting and offloading infrastructure to RPC providers and the ZKsync Node.

Starting Wednesday, July 10, we will enforce a per-IP rate limit of 20 requests per second (RPS) on Main Node, with bursts allowed up to 200 RPS. Stricter limits may also be applied to the `eth_getLogs` endpoint. This ensures that essential transactions, such as those from Metamask, are not disrupted while reducing API overuse.

We expect this rate limit will not affect most developers, partners, and clients. Regular access to our Main Node should remain unaffected. If your operations might be impacted, we recommend using the [ZKsync Node](https://docs.zksync.io/zksync-node) or an [RPC provider](https://docs.zksync.io/ecosystem/node-providers) for applications running on the ZKsync network.

**Comment by agolajko** [comment]
# Websocket Connection Update
In addition to our scheduled updates, we're implementing the following changes to WS:

- New rate limit: 1 new websocket connection per second
(Short bursts up to 10 connections/second allowed)
- Reminder: Existing limit of 1000 requests/minute per connection remains in effect
